src/emulator/functionsEmulator.ts

```diff
@@ -594,6 +638,10 @@ export class FunctionsEmulator implements EmulatorInstance {
     // Remove any old trigger definitions
     const toRemove = Object.keys(this.triggers).filter((recordKey) => {
       const record = this.getTriggerRecordByKey(recordKey);
+      if (record.backend.codebase !== emulatableBackend.codebase) {
+        // We are only loading one codebase, don't delete triggers from another.
+        return false;
+      }
```

Comment:
Not sure either way, but should this be below the `if (force)` conditional?

Reply:
No. The order is important here, and we should not be removing triggers from a different backend. This is what was happening before: If you had 2 codebases, then you loop through the codebases with loadTriggers and force=true. So codebaseA would delete all existing triggers, then add trigger1, trigger2 say, but then during the next loop, loadTriggers(codebaseB, true) would be run and it would delete trigger1, trigger2 and then add it's own triggers. So it clobbered the triggers that came before. BUT it wasn't too big of a problem because the eventarc remove was broken and the removeEventTriggers were not really implemented. So stuff stuck around by accident. It's fixed to work properly now: each codebase is only allowed to optionally remove its own triggers. (So the fixed eventarc deletion will not cause problems.)